src/main/java/de/tum/cit/aet/artemis/hyperion/dto/quiz/AiQuizGenerationResponseDTO.java

```diff
@@ -0,0 +1,8 @@
+package de.tum.cit.aet.artemis.hyperion.dto.quiz;
+
+import java.util.List;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+public record AiQuizGenerationResponseDTO(List<GeneratedMcQuestionDTO> questions, @JsonInclude(JsonInclude.Include.NON_EMPTY) List<String> warnings) {
```

Comment:
```suggestion
@JsonInclude(JsonInclude.Include.NON_EMPTY)
public record AiQuizGenerationResponseDTO(@NotNull List<GeneratedMcQuestionDTO> questions, @JsonInclude(JsonInclude.Include.NON_EMPTY) List<String> warnings) {
```
you need to annotate all DTOS with json.nonempty thats why you failing the architecture test: server-style
and add the proper annotation to parameters that are not of primitive types. Look at the suggestion

Reply:
👍